src/analyses/mutexAnalysis.ml

```diff
@@ -23,92 +20,42 @@ end
 (** Mutex analyzer without base --- this is the new standard *)
 module MakeSpec (P: SpecParam) =
 struct
-  include Analyses.DefaultSpec
-
-  (** name for the analysis (btw, it's "Only Mutex Must") *)
+  module Arg =
+  struct
+    module D = Lockset
+    module G = P.G
+    module V = VarinfoV
+
+    let add ctx l =
+      D.add l ctx.local
+
+    let remove ctx l =
+      D.remove (l, true) (D.remove (l, false) ctx.local)
+
+    let remove_all ctx =
+      (* TODO: implement in privatizations? *)
+      (* Mutexes.iter (fun m ->
+          ctx.emit (MustUnlock m)
+        ) (D.export_locks ctx.local); *)
```

Comment:
There is the question how privatizations should handle unlocking of unknown pointers, i.e. the must lockset essentially becoming empty. We haven't handled this before, but now there's the possibility.

I wouldn't want to try implementing a `unlock_all` for all of our privatizations right away, so maybe just have an unsoundness message in this case?

Reply:
>  so maybe just have an unsoundness message in this case?

Yes, I think that should suffice for now.